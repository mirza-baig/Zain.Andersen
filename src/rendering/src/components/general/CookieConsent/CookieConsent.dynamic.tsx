// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { LinkField, withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import { useEffect, useState } from 'react';
import classNames from 'classnames';
// Components
import { Component } from 'src/helpers/Component';
import { CookieConsentTheme } from './CookieConsent.theme';
import { ComponentProps } from 'lib/types/component-props';
import { Headline } from 'src/helpers/Headline';
import { BodyCopy } from 'src/helpers/BodyCopy';
import { SvgIcon } from 'src/helpers/SvgIcon';
import { getCookie, setCookie } from 'lib/utils/client-storage-utils';
import ButtonPrimary from 'src/helpers/Button/buttons/btn--primary';
import SingleButton from 'src/helpers/SingleButton/SingleButton';
import { Item } from '@sitecore-jss/sitecore-jss-nextjs';

export type CookieConsentProps =
  Feature.EnterpriseWeb.Enterprise.Components.General.CookieConsent.CookieConsent & ComponentProps;
const CookieConsent = (props: CookieConsentProps) => {
  const { themeData } = useTheme(CookieConsentTheme);
  const consentCookieName = 'privacy-notification';

  const [isDismissed, setIsDismissed] = useState(false);

  useEffect(() => {
    if (getCookie(consentCookieName)) {
      setIsDismissed(true);
    }
  }, []);

  if (isDismissed) {
    return null;
  }

  const setConsent = () => {
    setCookie(
      consentCookieName,
      1,
      new Date(new Date().getTime() + 1000 * 60 * 60 * 24 * 365).toUTCString()
    );

    setIsDismissed(true);
  };

  return (
    <Component variant="full" dataComponent="general/cookieconsent" {...props}>
      <div
        className={classNames(
          'theme-black fixed bottom-0 left-0 right-0 z-10 col-span-12 place-items-center bg-theme-bg'
        )}
      >
        <div className={classNames(themeData.classes.bannerWrapper)}>
          <div className={themeData.classes.textWrapper}>
            <Headline classes={themeData.classes.headline} {...props} />
            <BodyCopy classes={themeData.classes.bodyClass} {...props} />
          </div>
          <div className="col-span-2 place-content-center ml:col-span-3 ml:place-self-end ml:self-center">
            <div className={themeData.classes.buttonsWrapper}>
              <div
                onClick={(e) => {
                  e.preventDefault();
                  setConsent();
                }}
              >
                <ButtonPrimary
                  classes={themeData.classes.acceptCookieButton}
                  variant={
                    {
                      id: '',
                      name: '',
                      url: '',
                      fields: {
                        Value: {
                          value: 'primary',
                        },
                      },
                    } as Item
                  }
                  field={
                    {
                      value: {
                        href: '',
                        text: 'Accept',
                        linktype: 'external',
                        url: '',
                        anchor: '',
                        target: '_blank',
                      },
                    } as LinkField
                  }
                />
              </div>
              <SingleButton classes={themeData.classes.secondButton.classes} {...props} />
            </div>
          </div>
          <button
            className={classNames(themeData.classes.iconWrapper)}
            onClick={() => setConsent()}
            title="btn-close"
          >
            <SvgIcon icon="close" size="md" />
          </button>
        </div>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<CookieConsentProps>(CookieConsent);
