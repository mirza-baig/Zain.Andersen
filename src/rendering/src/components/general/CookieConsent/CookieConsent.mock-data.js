const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'CookieConsent',
  },
  fields: {
    acceptButtonLabel: {
      value: 'Accept',
    },
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    headlineLevel: null,
    headlineText: {
      value:
        'Cookies help us improve your website experience. By using our website, you agree to our use of cookies.',
    },
    body: {
      value: 'Lorem ipsum dolor amit.',
    },
    cta1Link: {
      value: {
        href: '',
      },
    },
    cta1ModalLinkText: {
      value: 'Learn More',
    },
    cta1Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: {
        Value: {
          value: 'arrow',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta1Modal: {
      id: '1edeac52-72fb-49a9-9a72-2917975e2f76',
      url: '/Data/Components/Generic-Modals/Generic-Modal-Large',
      name: 'Generic Modal Large',
      displayName: 'Generic Modal Large',
      fields: {
        desktopAnimationStyle: null,
        displayCloseBar: {
          value: false,
        },
        mobileAnimationStyle: null,
        modalSize: {
          id: 'b6bf33e1-366c-439e-9082-b9f3871ab5e3',
          url: 'http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Generic-Modal-Size/Large',
          name: 'Large',
          displayName: 'Large',
          fields: {
            Value: {
              value: 'large',
            },
          },
          templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
          templateName: 'Enum',
        },
        componentSpacing: null,
        sectionId: {
          value: '',
        },
        modalId: {
          value: 'genericModal2',
        },
        eventName: {
          value: '',
        },
        eventType: {
          value: '',
        },
        eventZone: {
          value: '',
        },
      },
      templateId: '1a9db7e8-eb55-41d1-88e2-a5d1f548b045',
      templateName: 'Generic Modal',
    },
    cta1Style: {
      id: 'dd818850-ec95-4d32-9774-7cc8173e277b',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Link',
      name: 'Link',
      displayName: 'Link',
      fields: {
        Value: {
          value: 'link',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
  },
};

export default defaultData;
