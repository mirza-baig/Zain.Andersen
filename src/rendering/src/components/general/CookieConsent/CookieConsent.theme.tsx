// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const CookieConsentTheme: ThemeFile = {
  aw: {
    classes: {
      headline: 'text-white text-s ml:text-m font-heavy',
      bannerWrapper:
        'grid-rows-auto grid grid-cols-2 place-items-center px-m ml:px-l lg:px-m ml:mx-auto ml:max-w-screen-lg min-h-[75px]  ml:grid-cols-12 ml:place-items-start my-s',
      bodyClass: 'text-white text-body font-regular mt-xxs',
      buttonsWrapper:
        'flex flex-col ml:flex-row justify-center items-center font-heavy mb-0 gap-y-6 ml:gap-0',
      acceptCookieButton: 'mr-s',
      secondButton: {
        classes: {
          wrapper: '!mb-0',
          cta1Classes: '',
        },
      },
      iconWrapper: 'absolute top-3 right-3 text-white',
      textWrapper: 'col-span-2 text-center ml:col-span-9 ml:text-left mb-m ml:mb-0 self-center',
    },
  },
  rba: {
    classes: {
      headline: 'text-xs text-white font-heavy ml:font-bold !font-serif',
      bannerWrapper:
        'grid-rows-auto grid grid-cols-2 px-m ml:px-l lg:px-m ml:mx-auto ml:max-w-screen-lg ml:grid-cols-12 place-items-start min-h-[48px] my-s ml:my-xs',
      bodyClass: 'text-white font-regular mt-xxs ml:mt-0 [&_.body-copy]:!text-sm-xxs',
      buttonsWrapper:
        'flex flex-col ml:flex-row ml:items-center font-bold gap-y-6 gap-x-6 ml:space-x-0 mb-xxs ml:mb-0',
      acceptCookieButton: 'mr-0 px-s',
      secondButton: {
        classes: {
          wrapper: '!mb-0 text-white',
          cta1Classes: '',
        },
      },
      iconWrapper: 'absolute top-3 right-3 text-white',
      textWrapper: 'col-span-2 ml:col-span-9 mb-s ml:mb-0 self-center ',
    },
  },
};
