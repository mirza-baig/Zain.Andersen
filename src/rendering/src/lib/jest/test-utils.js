// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
