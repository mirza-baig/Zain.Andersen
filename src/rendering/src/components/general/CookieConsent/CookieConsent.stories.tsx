// Global
import { Story, Meta } from '@storybook/react';
// Local
import CookieConsent, { CookieConsentProps } from './CookieConsent.dynamic';
import defaultData from './CookieConsent.mock-data';

export default {
  title: 'General/CookieConsent',
  component: CookieConsent,
} as Meta;
const Template: Story<CookieConsentProps> = (props) => <CookieConsent {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
