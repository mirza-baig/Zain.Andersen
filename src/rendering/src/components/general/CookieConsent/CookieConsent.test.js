// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import CookieConsent from './CookieConsent.dynamic';
import defaultData from './CookieConsent.mock-data';

it('renders correctly', async () => {
  const component = snapshot(CookieConsent, { componentProps: defaultData });
  await hasDataComponent(component, 'general/cookieconsent');
});
