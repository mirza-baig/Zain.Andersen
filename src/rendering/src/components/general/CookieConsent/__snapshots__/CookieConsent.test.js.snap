// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/cookieconsent"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="theme-black fixed bottom-0 left-0 right-0 z-10 col-span-12 place-items-center bg-theme-bg"
      >
        <div
          class="grid-rows-auto grid grid-cols-2 place-items-center px-m ml:px-l lg:px-m ml:mx-auto ml:max-w-screen-lg min-h-[75px]  ml:grid-cols-12 ml:place-items-start my-s"
        >
          <div
            class="col-span-2 text-center ml:col-span-9 ml:text-left mb-m ml:mb-0 self-center"
          >
            <div
              class="items-top font-sans [&_a:hover]:underline text-white text-s ml:text-m font-heavy"
            >
              <h2>
                Cookies help us improve your website experience. By using our website, you agree to our use of cookies.
              </h2>
            </div>
            <div
              class="text-white text-body font-regular mt-xxs"
            >
              <div
                class="body-copy"
              >
                <div>
                  Lorem ipsum dolor amit.
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-span-2 place-content-center ml:col-span-3 ml:place-self-end ml:self-center"
          >
            <div
              class="flex flex-col ml:flex-row justify-center items-center font-heavy mb-0 gap-y-6 ml:gap-0"
            >
              <div>
                <a
                  aria-label="Accept (Opens in a new tab)"
                  class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed mr-s"
                  data-gtm-click=""
                  data-gtm-dl-event="cta_click"
                  href=""
                  rel="noopener noreferrer"
                  role="link"
                  tabindex="0"
                  target="_blank"
                >
                  Accept
                  <span
                    class="sr-only"
                  >
                     (Opens in a new tab)
                  </span>
                </a>
              </div>
              <div
                class="!mb-0 mb-s flex items-start md:flex-row md:items-center"
              >
                <button
                  class="hover:decoration-theme-btn-decoration flex w-fit items-center whitespace-normal font-sans text-text-link font-heavy text-theme-text hover:underline hover:underline-offset-8 disabled:border-gray disabled:text-gray"
                >
                  Learn More
                   
                  <span
                    class="ml-xxs"
                  />
                </button>
              </div>
            </div>
          </div>
          <button
            class="absolute top-3 right-3 text-white"
            title="btn-close"
          >
            <span />
          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;
